curriculum/challenges/english/25-front-end-development/lab-fcc-forum-leaderboard/673c91f0b934834bc4a3ecc2.md

```diff
@@ -0,0 +1,1422 @@
+  ],
+  topic_list: {
+    topics: [
+      {
+        bumped_at: "2024-04-15T16:01:26.403Z",
+        category_id: 1,
+        id: 684569,
+        posters: [{user_id: 6}, {user_id: 170865}, {user_id: 285941}],
+        posts_count: 8,
+        slug: "the-freecodecamp-podcast-is-back-now-with-video",
+        title: "The freeCodeCamp Podcast is back – now with video",
+        views: 542
+      },
+      {
+        bumped_at: "2024-04-19T13:52:03.523Z",
+        category_id: 421,
+        id: 686149,
+        posters: [{user_id: 170865}],
+        posts_count: 1,
+        slug: "problem-with-making-changes-to-styles-js",
+        title: "Problem with making changes to styles. (JS)",
+        views: 9
+      }
+    ]
+  }
+}
+const pContainer = document.getElementById("posts-container");
+pContainer.innerHTML = "";
+showLatestPosts(data);
+
+assert.equal(pContainer.querySelector("tr:first-child>td:nth-child(3)").innerText, "7");
+
+assert.equal(pContainer.querySelector("tr:last-child>td:nth-child(3)").innerText, "0");
+```
+
+The fourth `td` element of each table row from the string returned by `showLatestPosts` should contain the number of views of the post.
+
+```js
+const data = {
+  users: [
+    {
+      avatar_template: "/user_avatar/forum.freecodecamp.org/quincylarson/{size}/212400_2.png",
+      id: 6,
+      name: "Quincy Larson",
+      username: "QuincyLarson"
+    },
+    {
+      avatar_template: "/user_avatar/forum.freecodecamp.org/jwilkins.oboe/{size}/179497_2.png",
+      id: 285941,
+      name: "Jessica Wilkins",
+      username: "jwilkins.oboe"
+    },
+    {
+      avatar_template: "/user_avatar/forum.freecodecamp.org/ilenia/{size}/270648_2.png",
+      id: 170865,
+      name: "Ilenia",
+      username: "ilenia"
+    }
+  ],
+  topic_list: {
+    topics: [
+      {
+        bumped_at: "2024-04-15T16:01:26.403Z",
+        category_id: 1,
+        id: 684569,
+        posters: [{user_id: 6}, {user_id: 170865}, {user_id: 285941}],
+        posts_count: 8,
+        slug: "the-freecodecamp-podcast-is-back-now-with-video",
+        title: "The freeCodeCamp Podcast is back – now with video",
+        views: 542
+      },
+      {
+        bumped_at: "2024-04-19T13:52:03.523Z",
+        category_id: 421,
+        id: 686149,
+        posters: [{user_id: 170865}],
+        posts_count: 1,
+        slug: "problem-with-making-changes-to-styles-js",
+        title: "Problem with making changes to styles. (JS)",
+        views: 9
+      }
+    ]
+  }
+}
+const pContainer = document.getElementById("posts-container");
+pContainer.innerHTML = "";
+showLatestPosts(data);
+
+assert.equal(pContainer.querySelector("tr:first-child>td:nth-child(4)").innerText, "542");
+
+assert.equal(pContainer.querySelector("tr:last-child>td:nth-child(4)").innerText, "9");
+```
+
+The fifth `td` element of each table row from the string returned by `showLatestPosts` should contain time passed since the last activity, generated using the `timeAgo` function.
```

Comment:
I had this failing because I was using `round` instead of `floor`, it's difficult to see why it's failing

Reply:
I added two tests. Let me know if it can work like that